source/includes/steps-fle-create-data-encryption-key.yaml

```diff
@@ -11,19 +11,17 @@ content: |
         :tabid: java-sync
 
         .. code-block:: java
-           :emphasize-lines: 11
+           :emphasize-lines: 6
 
            String path = "master-key.txt";
 
-           byte[] fileBytes = new byte[96];
+           byte[] localMasterKey= new byte[96];
 
            try (FileInputStream fis = new FileInputStream(path)) {
-             fileBytes = fis.readAllBytes();
-           } catch (IOException e) {
-             e.printStackTrace();
+               fis.readNBytes(localMasterKey, 0, 96);
+           } catch (Exception e) {
+               e.printStackTrace();
```

Comment:
When you put all the examples together, this catch is problematic, as it swallows the error and proceeds to the next line.  Consider removing the catch clause altogether.

Reply:
Would it compile without `catch` and no `finally`? I was getting lint errors in Eclipse running JDK 1.8.

Reply:
Hmm, haven't tried with java 8, but you shouldn't need either with try-with-resources

Reply:
Looks like it was introduced in 1.7, but since the driver is compatible with 1.6, would you recommend I use it anyway? Or should I find another way to structure this block?

Reply:
We required Java 8 for FLE so consider Java 8 as the minimum you need to be compatible with.

Reply:
Good to know, will add that as an explicit requirement in the guide.

Reply:
I still receive the same compile errors whether setting compliance to 1.8 or 12. Am I doing something incorrectly?

```
Unhandled exception type IOException thrown by automatic close() invocation on fis
	Unhandled exception type FileNotFoundException
	Unhandled exception type IOException
```

Reply:
That's expected.  Just add `IOException` to the throws clause of the containing method.

Reply:
The containing method in the context of the code samples is `main`.  Even if it was in another function, there would need to be a catch statement somewhere. If I added a `System.exit(1)` call after the stacktrace, would that at least meet the requirement of halting on an exception there?

Reply:
This is legal:

    public static void main(String[] args) throws IOException

Reply:
Ah, you got me there. Will update.